File: Codelearn.io/Training/C++/isValidDate.cpp
bool isValidDate(int d, int m, int y)
{
    int c = 1;
    switch (m) {
        case 1: case 3: case 5: case 7: case 8: case 10: case 12:
            if (d < 1 || d > 31) c = 0;
            break;

        case 4: case 6: case 9: case 11:
            if (d < 1 || d > 30) c = 0;
            break;

        case 2:
            if ((y % 4 == 0 && y % 100 != 0) || y % 400 == 0) {
                if (d < 1 || d > 29) c = 0;
            } else {
                if (d < 1 || d > 28) c = 0;
            }
            break;
        default: c = 0; break;
    }
    if (y < 1) c = 0;
	return c;
}

File: Codelearn.io/Training/Python/isValidDate.py
def isValidDate(d, m, y):
    c = 1
    if y < 1: return False
    if m in [1, 3, 5, 7, 8, 10, 12]: return not (d < 1 or d > 31)
    elif m in [4, 6, 9, 11]: return not (d < 1 or d > 30)
    elif m == 2:
        if (y % 4 == 0 and y % 100 != 0) or y % 400 == 0: return not (d < 1 or d > 29)
        else: return not (d < 1 or d > 28)
    else: return False